WordPress/gutenberg
# Flex: Convert component to TypeScript

## What?
Converts the `Flex`, `FlexItem` and `FlexBlock` components to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactors the current `Flex`, `FlexItem` and `FlexBlock` to TypeScript.
- Refactors the tests to TypeScript.
- Refactors the storybook to TypeScript.

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `Flex` continues to function as expected
3. Test the Gutenberg editor that everything works as before.
